src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -210,15 +263,22 @@ private static bool SymbolIsStaticAndAbstract(ISymbol symbol, ConcurrentDictiona
         {
             ISymbol symbol = context.Symbol;
 
-            // Don't report diagnostics on symbols that are marked as Preview
-            if (ProcessPreviewAttribute(symbol, requiresPreviewFeaturesSymbols, previewFeatureAttributeSymbol))
+            // Don't report diagnostics on symbols that are marked as Preview or if the ContainingType is Preview. We'll miss 2+ level nested type/method definitions, but that's ok. Only users who suppress the analyzer will run into this case.
```

Comment:
While suppression of this analyzer is fine. I would expect the correct thing to do on the users part is also mark their types as preview. Becasue of this I would expect us to want to always find all locations

Reply:
I think this does find all locations right? All this (and the next) if block is doing is:

```csharp
    [RequiresPreviewFeatures]
    public class MarkedClass // Don't report a diagnostic here
    {
    }
```

When `MarkedClass` is being processed, we detect that it is already marked with `[RequiresPreviewAttribute]` and return. The next if block check the immediate parent when type members of Preview classes are processed. SOmething like:

```csharp
    [RequiresPreviewFeatures]
    public class MarkedPreviewClass
    {

        public MarkedPreviewClass() { } // Don't report a diagnostic here

        public void UnmarkedMethodInPreviewClass() { } // Don't report a diagnostic here

    }
```

Reply:
It's not. We now examine the containing symbol chain all the way to the assembly. Deleted the comment